e2e/avpFixtures.js

```diff
@@ -0,0 +1,97 @@
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+/**
+ * avpFixtures.js
+ *
+ * @file This module provides custom fixtures specifically tailored for Accessibility, Visual, and Performance (AVP) tests.
+ * These fixtures extend the base functionality of the Playwright fixtures and appActions, and are designed to be
+ * generalized across all plugins. They offer functionalities like scanning for accessibility violations, integrating
+ * with axe-core, and more.
+ *
+ * IMPORTANT NOTE: This fixture file is not intended to be extended further by other fixtures. If you find yourself
+ * needing to do so, please consult the documentation and consider creating a specialized fixture or modifying the
+ * existing ones.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const { test, expect } = require('./pluginFixtures');
+const AxeBuilder = require('@axe-core/playwright').default;
+
+// Constants for repeated values
+const TEST_RESULTS_DIR = './test-results';
+
+/**
+ * Scans for accessibility violations on a page and writes a report to disk if violations are found.
+ * Automatically asserts that no violations should be present.
+ *
+ * @typedef {object} GenerateReportOptions
+ * @property {string} [reportName] - The name for the report file.
+ *
+ * @param {import('playwright').Page} page - The page object from Playwright.
+ * @param {string} testCaseName - The name of the test case.
+ * @param {GenerateReportOptions} [options={}] - The options for the report generation.
+ *
+ * @returns {Promise<object|null>} Returns the accessibility scan results if violations are found,
+ *                                  otherwise returns null.
+ */
+exports.scanForA11yViolations = async function (page, testCaseName, options = {}) {
+  const builder = new AxeBuilder({ page });
+  builder.withTags(['wcag2aa']);
+  // https://github.com/dequelabs/axe-core/blob/develop/doc/rule-descriptions.md
+  builder.disableRules(['color-contrast']);
+  const accessibilityScanResults = await builder.analyze();
+
+  // Assert that no violations should be present
+  expect(
+    accessibilityScanResults.violations,
+    `Accessibility violations found in test case: ${testCaseName}`
+  ).toEqual([]);
+
+  // Check if there are any violations
+  if (accessibilityScanResults.violations.length > 0) {
+    let reportName = options.reportName || testCaseName;
+    let sanitizedReportName = reportName.replace(/\//g, '_');
+    const reportPath = path.join(TEST_RESULTS_DIR, `${sanitizedReportName}.json`);
+
+    try {
+      if (!fs.existsSync(TEST_RESULTS_DIR)) {
```

Comment:
i think we should be using `fs/promises` here instead. Also, we shouldn't check for the file's existence before creating the directory as we could get into a [race condition](https://nodejs.org/docs/latest-v18.x/api/fs.html#fs_fs_exists_path_callback). something like this instead:
```js
const fs = require('fs').promises;

async function writeAccessibilityReport(reportPath, accessibilityScanResults) {
    try {
        try {
            await fs.mkdir(TEST_RESULTS_DIR, { recursive: true });
        } catch (error) {
            if (error.code !== 'EEXIST') {
                throw error; // Throw the error if it is not because the directory already exists
            }
        }
        
        const data = JSON.stringify(accessibilityScanResults, null, 2);
        await fs.writeFile(reportPath, data); 
        console.log(`Accessibility report with violations saved successfully as ${reportPath}`);
        return accessibilityScanResults;
    } catch (err) {
        console.error(`Error writing the accessibility report to file ${reportPath}:`, err);
        throw err;
    }
}
```